src/query/storage/m3/multi_fetch_tags_result.go

```diff
@@ -23,19 +23,60 @@ package m3
 import (
 	"sync"
 
-	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/dbnode/client"
 	xerrors "github.com/m3db/m3x/errors"
 )
 
-type multiFetchTagsResult struct {
+const initSize = 10
+
+type multiSearchResult struct {
 	sync.Mutex
-	result    *storage.SearchResults
+	seenIters []client.TaggedIDsIterator // track known iterators to avoid leaking
```

Comment:
again, seems like you need a builder here to avoid the need to have FinalResult()

Reply:
I'm going to leave this one for now so it more closely matches multi_fetch_result.go, but this is essentially a builder under a different name

Reply:
ok